pkg/traceability/sampling/globalsampling.go

```diff
@@ -156,6 +187,75 @@ func FilterEvents(events []publisher.Event) ([]publisher.Event, error) {
 	return agentSamples.FilterEvents(events), nil
 }
 
+func RemoveApiAppKey(apiServiceName, managedAppName string) {
+	if agentSamples != nil && agentSamples.apiAppErrorSampling != nil {
+		externalAPIID := getExternalAPIID(apiServiceName)
+		externalAppID := getExternalAppID(managedAppName, getExternalAppKeyData())
+		k := externalAPIID + externalAppID
+
+		agentSamples.samplingLock.Lock()
+		defer agentSamples.samplingLock.Unlock()
+		delete(agentSamples.apiAppErrorSampling, k)
+	}
+}
+
+func getExternalAppID(appName string, externalAppKey definitions.ExternalAppData) string {
+	cacheManager := getCacheManager()
```

Comment:
do we need to get the cachemanger every time or can we set this as an attribute on the sampling object?